nano/node/confirmation_height_processor.cpp

```diff
@@ -0,0 +1,426 @@
+#include <nano/node/confirmation_height_processor.hpp>
+
+#include <boost/optional.hpp>
+#include <cassert>
+#include <nano/lib/logger_mt.hpp>
+#include <nano/lib/numbers.hpp>
+#include <nano/lib/utility.hpp>
+#include <nano/node/active_transactions.hpp>
+#include <nano/node/stats.hpp>
+#include <nano/secure/blockstore.hpp>
+#include <nano/secure/common.hpp>
+#include <nano/secure/ledger.hpp>
+#include <numeric>
+
+nano::confirmation_height_processor::confirmation_height_processor (nano::pending_confirmation_height & pending_confirmation_height, nano::block_store & store, nano::ledger & ledger, nano::active_transactions & active, nano::logger_mt & logger) :
+pending_confirmations (pending_confirmation_height),
+store (store),
```

Comment:
I guess store can be received from ledger.store without extra reference

Reply:
Looking again I don't actually need the `ledger` reference. It was only used to access the stats (`ledger.stats`) which I am now passing in directly and to call `ledger.account` which is just a wrapper around `store.block_account`. So this removes a rather large dependency.